docs/developer.md

```diff
@@ -0,0 +1,75 @@
+
+You can generate additional fixtures with statements like `python manage.py dumpdata theses.Person --pks=63970,29903 > hamlet/theses/fixtures/authors.json`, but make sure to include the pks of all objects already in the fixtures (or to write it to a separate file and then unite it with the existing - you can't just append because the json syntax will be wrong). Also make sure that the theses you use are in fact present in the test neural net.
+
+### Checking that a document is in a given neural net
+
+* Make sure your settings file points to the desired `MODEL_FILE`
+* `python manage.py shell`
+
+```
+from gensim.models.doc2vec import Doc2Vec
+from django.conf import settings
+model = Doc2Vec.load(settings.MODEL_FILE)
+identifier = '1721.1-%d.txt' % YOUR THESIS IDENTIFIER HERE
+identifier in model.docvecs.doctags.keys()
+```
+
+If you don't have a target thesis object but you need one you know is in the neural net, look at the output of `model.docvecs.doctags.keys()`. This is a list of filenames of text files from dspace; they are all of the format `1721.1-NUMBER.txt`, where `NUMBER` is the identifier of the thesis. You can look up `Thesis` objects in your database by this identifier (which is `Thesis.identifier`, not the primary key).
+
+## System configuration
+
+### Development dependencies: pipenv
+For the most part, dependencies are installed via pipenv. There's a `.env` file (kept out of version control) for use by `pipenv shell`. It specifies:
+* `DJANGO_SETTINGS_MODULE`
+  * `DJANGO_SETTINGS_MODULE='hamlet.settings.local'` (for using heroku local)
+  * `DJANGO_SETTINGS_MODULE='hamlet.settings.base'` (for python manage.py runserver)
+* `DJANGO_DB_PASSWORD='(your password)'`
+* `DJANGO_DEBUG_IS_TRUE='True' (if you want)`
+* `DSPACE_OAI_IDENTIFIER`
+* `DSPACE_OAI_URI`
+
+The latter two are only relevant if you plan to be downloading files or metadata from DSpace. They can be omitted or given dummy values otherwise.
+
+### Additional non-pipenv dependencies
+Some dependencies require extra help:
+* tika requires Java
+* nltk may require installing corpora through the python shell
+* gensim wants a C compiler (it can run without one but will be 70x slower; a single neural net training run can take literally days in this case)
+* python-magic needs libmagic (`brew install libmagic` on OSX).
+* captcha says it needs `apt-get -y install libz-dev libjpeg-dev libfreetype6-dev python-dev` or similar. You can't yum install them on AWS, but the captcha works anyway, so maybe it's lying.
+
+You only need the first three of these if you plan to be doing neural net training. If you're developing the Django parts you can skip them; just get a prebuilt neural net file (see below, "Neural net files").
+
+### Other config
+Postgres needs a database and user (default values are `hamlet` for database name and username, no password; override this if desired in `.env` with `DJANGO_DB`, `DJANGO_DB_USER`, `DJANGO_DB_PASSWORD`)
```

Comment:
One suggestion would be to also add the ENGINE setting to .env. I'm actually running things locally with sqlite since it's easier to set up in dev. I'm running things with a custom setting file in order to do this, but would be nice to be able to skip that and put the config in .env.

Reply:
Good idea; done and documented.